source/io/JavaScriptInvoke.cpp

```diff
@@ -0,0 +1,123 @@
+/**
+
+Copyright (c) 2018 National Instruments Corp.
+
+This software is subject to the terms described in the LICENSE.TXT file
+
+SDG
+*/
+
+/*! \file
+\brief Functions to call user defined JavaScript functions
+*/
+
+#include "TypeDefiner.h"
+#include "ExecutionContext.h"
+#include "StringUtilities.h"
+#include "TDCodecVia.h"
+#include <stdio.h>
+
+#if defined (VIREO_TYPE_JavaScriptInvoke)
+namespace Vireo {
+
+#if kVireoOS_emscripten
+extern "C" {
+    // JavaScript function prototypes
+    // Parameters: functionName, parameters*, parametersCount, errorCheckingEnabled, errorStatus*, errorCode*, errorSource*
+    extern void jsJavaScriptInvoke(StringRef, void *, Int32, Boolean, Boolean *, Int32 *, StringRef);
+}
+#endif
+
+// Keep in synchrony with JAVASCRIPT_PARAMETER_TYPE in module_javaScriptInvoke.js
+enum JavaScriptParameterTypeEnum {
+    kJavaScriptParameterType_None = 0,
+    kJavaScriptParameterType_Int32,
+    kJavaScriptParameterType_UInt32,
+    kJavaScriptParameterType_String,
+    kJavaScriptParameterType_Boolean
+};
+
+//------------------------------------------------------------
+//! Return parameter type for the given element(index) in the parameters array
+VIREO_EXPORT Int32 Data_GetParameterType(StaticTypeAndData *parameters, Int32 index)
```

Comment:
The Data_* exports are meant to be fairly generic for the C API of talking to Vireo and are in EggShell.
These functions look very specific to JSLI. I think they should not be called Data_*.

Maybe something to emphasize they are JSLI specific like JavaScriptInvoke_GetParameterType or JavaScriptInvoke_GetParameterPointer.